src/iter.rs

```diff
@@ -1,20 +1,24 @@
 use crate::progress::ProgressBar;
 use std::io::{self, IoSliceMut};
 use std::iter::FusedIterator;
+use std::convert::TryFrom;
 
 /// Wraps an iterator to display its progress.
 pub trait ProgressIterator
 where
     Self: Sized + Iterator,
 {
-    /// Wrap an iterator with default styling. Attempt to guess iterator
-    /// length using `Iterator::size_hint`.
-    fn progress(self) -> ProgressBarIter<Self> {
-        let n = match self.size_hint() {
-            (_, Some(n)) => n as u64,
-            _ => 0,
-        };
-        self.progress_count(n)
+    /// Wrap an iterator with default styling. Uses `Iterator::size_hint` to get length.
+    /// Returns `Some(..)` only if `size_hint.1` is `Some`. If you want to create a progress bar
+    /// even if `size_hint.1` returns `None` use `progress_count` or `progress_with` instead.
```

Comment:
Do you actually have a use case for this (or have you seen anyone else express a use case for it)? If not, I'd rather wait and see if someone complains about the new behavior. We can always add it later in a compatible release...